site/src/pages/release-notes/1.18.0.mdx

```diff
@@ -0,0 +1,134 @@
+---
+date: 2022-08-05
+---
+
+## 🌟 New features
+
+- <type://DocService> can now render Markdown and [Mermaid](https://mermaid-js.github.io/mermaid/)
+formatted documentation for annotated services #4299 #4304
+
+```java Markdown
+@Description(value = "### Markdown format", markup = Markup.MARKDOWN)
+@Get("/markdown")
+public HttpResponse markdown() {
+    return HttpResponse.of(200);
+}
+```
+
+```java Mermaid
+@Description(value = "graph TD;\n" +
+                     " A-->B;\n" +
+                     " A-->C;\n" +
+                     " B-->D;\n" +
+                     " C-->D;",
+             markup = Markup.MERMAID)
+@Get("/mermaid")
+public HttpResponse mermaid() {
+    return HttpResponse.of(200);
+}
+```
+
+- It is now possible to bind decorators to gRPC services using <type://GrpcServiceBuilder>. #4311 #4316
+```java
+GrpcService.builder()
+           .addService(myGrpcService, List.of(LoggingService.newDecorator()))
+           .build();
+```
+
+- Armeria's gRPC users can now use an <type://UnframedGrpcErrorHandler> which provides a
+[richer error model](https://cloud.google.com/apis/design/errors#error_model) for unframed requests.
+
+```java
+Server.builder()
+      .service(GrpcService.builder()
+                          .addService(grpcService)
+                          .unframedGrpcErrorHandler(UnframedGrpcErrorHandler.ofRichJson())
+                          .build());
+```
+
+```json
+{
+	"error": {
+		"code": 2,
+		"message": "A custom message!"
+	}
+}
+```
+
+- Multipart file upload is now available for GraphQL users. #3971 #4199
+```java
+GraphqlService service =
+    GraphqlService.builder()
+...
+                  .runtimeWiring(c -> {
+                      c.scalar(MoreScalars.multipartFile());
+                      DataFetcher<String> upload = environment -> {
+                          MultipartFile multipartFile = environment.getArgument("file");
+                          ...
+                      }
+                      c.type("Multipart", typeWiring -> typeWiring.dataFetcher("upload", upload));
+                  })
+...
+```
+
+- Users can now extract the innermost element of <type://Unwrappable> type objects easily using <type://Unwrappable#unwrapAll()> #4338
+```java
+FooDecorator extends SimpleDecoratingHttpClient {
+...
+    @Override
+    public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
+        Object innermost = unwrapAll();
+        ...
+    }
+}
```

Comment:
Personally, I prefer the example used in the Javadoc of `unwrapAll()`. 😉
https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/util/Unwrappable.java#L96